pandas/core/reshape/merge.py

```diff
@@ -952,6 +975,53 @@ def _validate_specification(self):
         if len(self.right_on) != len(self.left_on):
             raise ValueError("len(right_on) must equal len(left_on)")
 
+    def _validate(self, validate):
+
+        # Get axes
+        left_key = self.left_on if self.left_on is not None else self.on
+        right_key = self.right_on if self.right_on is not None else self.on
+
+        # Check uniqueness of each
+        if self.left_index:
+            left_unique = not (self.orig_left.index.duplicated()).any()
+        else:
+            left_unique = not (self.orig_left[left_key].duplicated()).any()
+
+        if self.right_index:
+            right_unique = not (self.orig_right.index.duplicated()).any()
+        else:
+            right_unique = not (self.orig_right[right_key].duplicated()).any()
```

Comment:
@nickeubank the point is you are repeating lots of code for no reason. The notion of figuring out which is what (from the input args) is actually non-trivial and will be refactored shortly. Don't repeat it.